lint/unusedParams_checker.go

```diff
@@ -0,0 +1,55 @@
+package lint
+
+import (
+	"fmt"
+	"go/ast"
+)
+
+func init() {
+	addChecker(unusedParamsChecker{}, &ruleInfo{})
+}
+
+type unusedParamsChecker struct {
+	baseFuncDeclChecker
+}
+
+func (c unusedParamsChecker) New(ctx *context) func(*ast.File) {
+	return wrapFuncDeclChecker(&unusedParamsChecker{
+		baseFuncDeclChecker: baseFuncDeclChecker{ctx: ctx},
+	})
+}
+
+func (c *unusedParamsChecker) CheckFuncDecl(decl *ast.FuncDecl) {
+	params := decl.Type.Params
+	if params == nil || params.NumFields() == 0 {
+		return
+	}
+
+	objUsed := make(map[*ast.Object]struct{})
+	for id := range c.ctx.TypesInfo.Uses {
+		objUsed[id.Obj] = struct{}{}
+	}
+
+	for _, p := range params.List {
+		if len(p.Names) == 0 {
+			c.warnUnnamed(p)
+			break
+		}
+		for _, name := range p.Names {
```

Comment:
Hint: calling `p.Names` element as `id` (ident) makes `name.Name`=>`id.Name`, which is somewhat more pleasant.
Note that `Names` are `[]*ast.Ident`, so it does make sense.
(Optional.)

Reply:
ok